### ISSUE TITLE
No SSH access in Tumbleweed

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2767]
I just did a VM install of tumbleweed and I can't connect through SSH.

(...)
In that case the 'problem' is a discrepency between `PermitRootLogin yes` on a Leap install (not a fan long term, but important initially to get the keys transferred,) and `PermitRootLogin prohibit-password` on Tumbleweed.

_Originally posted by @RlndVt in https://github.com/rockstor/rockstor-core/issues/2277#issuecomment-1852836002_

**Comment by RlndVt** [comment]
Is this now a duplicate of #2765 ?

**Comment by FroggyFlox** [comment]
No. Please see my comment there (https://github.com/rockstor/rockstor-core/issues/2765#issuecomment-1858537267)... looks like it took me longer to finish writing that one that I thought... sorry for the confusion there.

**Comment by FroggyFlox** [comment]
Linking to a previous issue by @flukejones reporting the same behavior; interestingly, that was later changed upstream so it seems it has changed again since: #2005.

A PR fixing the current issue should also automatically close #2005.

**Comment by phillxnet** [comment]
@FroggyFlox Re:

> Adding `PermitRootLogin yes` to `/etc/ssh/sshd_config.d/rockstor-sftp.conf` and restarting the sshd service restores ssh access for root.

I can confirm that on a freshly built installer (today), profile "Tumbleweed.x86_64", the following equivalent configuration change restores our intended root ssh login behaviour (by password):
```
echo "PermitRootLogin yes" >> /etc/ssh/sshd_config.d/PermitRootLogin.conf systemctl reload sshd
systemctl reload sshd
```
Which is the same additional file found by default in a current Tumbleweed JeOS image as downloaded a few days ago:

> https://download.opensuse.org/ports/aarch64/tumbleweed/appliances/openSUSE-Tumbleweed-ARM-JeOS-efi.aarch64.qcow2

**Comment by phillxnet** [comment]
I've just updated the title on this issue as during pre-release testing of proposed new installers, the 15.6 (RC status upstream still) exhibits the same behaviour. With the same procedure to enable, re my last comment, working similarly. I.e. Leap 15.6 similarly requires the explicit `PermitRootLogin yes` configuration with follow-up sshd reload.

**Comment by phillxnet** [comment]
N.B. for the most recent build of our 15.6 installer profile `Leap15.6.x86_64` the resulting install now has working `root` user SSH password auth. This is as a result of upstrteam changes we for 15.6 we are inline with upstrteam.

Also note that our recent (pending release) TW installers, via upstream defaults, do not allow password auth root user SSH - so again we are in-line with upstream defaults as we do not override this.
